pkg/controllers/manila/manila.go

```diff
@@ -69,24 +72,61 @@ func NewManilaController(
 	eventRecorder events.Recorder) factory.Controller {
 
 	scInformer := informers.InformersFor("").Storage().V1().StorageClasses()
+	csiInformer := informers.InformersFor("").Storage().V1().CSIDrivers()
 	c := &ManilaController{
-		operatorClient:     operatorClient,
-		kubeClient:         kubeClient,
-		storageClassLister: scInformer.Lister(),
-		openStackClient:    openStackClient,
-		csiControllers:     csiControllers,
-		eventRecorder:      eventRecorder.WithComponentSuffix("ManilaController"),
+		operatorClient:       operatorClient,
+		kubeClient:           kubeClient,
+		storageClassLister:   scInformer.Lister(),
+		csiDriverClassLister: csiInformer.Lister(),
+		openStackClient:      openStackClient,
+		csiControllers:       csiControllers,
+		eventRecorder:        eventRecorder.WithComponentSuffix("ManilaController"),
 	}
 	return factory.New().WithSync(c.sync).WithSyncDegradedOnError(operatorClient).ResyncEvery(resyncInterval).WithInformers(
 		operatorClient.Informer(),
 		scInformer.Informer(),
+		csiInformer.Informer(),
 	).ToController("ManilaController", eventRecorder)
 }
 
+func (c *ManilaController) refreshCSIConfig(ctx context.Context) error {
+	csiConfigData := make(map[string]string)
+	fsGroupPolicy := "None"
+
+	if os.Getenv("CSI_FSGROUP_POLICY") != "" {
+		fsGroupPolicy = os.Getenv("CSI_FSGROUP_POLICY")
+	}
+
+	csiConfigData["fsGroupPolicy"] = fsGroupPolicy
+
+	csiConfig := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      util.ManilaCsiConfig,
+			Namespace: util.CloudConfigNamespace,
+		},
+		Data: csiConfigData,
+	}
+	_, _, err := resourceapply.ApplyConfigMap(ctx, c.kubeClient.CoreV1(), c.eventRecorder, &csiConfig)
```

Comment:
most attributes of `CSIDriver` object are immutable, so if configmap changes the fsGroup policy - `CSIDriver` has to be deleted and recreated. If `None` is what we want as default - why not specify in the manifest template above, rather than syncing it from configmap?

Reply:
Right, that field of the `CSIDriver` object is immutable (like the other spec fields), and if the `CSIDriver` object is deleted, as per this code it's recreated getting the `fsgroup` value defined in the `ConfigMap`.
This makes the `ConfigMap` an extra layer that is used to consume the config options we'd like to propagate to the driver, and establishes a pattern where the input are defined using the manifest (as done in the template above), but are collected in the `ConfigMap` that can be used for multiple purposes in a follow up work (e.g., extra logic in the `StorageClasses` sync).
However, if you think this layer is too much for now (e.g., more complexity for a little gain), we can rework the PR to remove the `ConfigMap` and sync it from the env.

Reply:
@gnufied  I'm going to push an extra change to simplify the code and avoid having this extra `ConfigMap` layer.

Reply:
I guess this part of the conversation can be now resolved. 

tl;dr: I removed the `ConfigMap`, and the `fsGroupPolicy` value is configured using the manifest env var section. Editing this field is something that can't be done as it's immutable (and not the goal of the PR); however, if the operator delete the `CSIDriver` resource and restart the operator with a different value, that object is recreated accordingly.

Reply:
There are several problems with approach proposed here:
1. You are using `ApplyCSIDriver` function - which will not allow you to change `fsGroupPolicy`, so once created `CSIDriver` object is immutable. 
2. 
3. Manila operator is a default Driver and I am not sure where will users edit the deployment template and specify `fsGroupPolicy` as environment variable. The changes to deployment template will be overridden by cluster-storage-operator.  This is not how customization of CVO managed operators work.

Reply:
That's not the point: the goal is not giving the opportunity to the (human) operator to edit an immutable field, but to expose an env variable for corner cases where people need to set a value != None.
If you delete the resource `CSIDriver`, it gets recreated/reconciled with the value specified as input.
To make people able to change this behavior, they should stop the operator, delete the resource and restart it with a different value, but definitely not a path we're supporting by default.

Reply:
> To make people able to change this behavior, they should stop the operator, delete the resource and restart it with a different value, but definitely not a path we're supporting by default.

If users disables cluster-storage-operator or set it to `Unmanaged`, so it does not overwrite Manila operator Deployment, then the cluster cannot be upgraded. This is good for local debugging, but not really usable in any production env.